textdb/textdb-dataflow/src/main/java/edu/uci/ics/textdb/dataflow/dictionarymatcher/DictionaryMatcher.java

```diff
@@ -0,0 +1,159 @@
+
+package edu.uci.ics.textdb.dataflow.dictionarymatcher;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import edu.uci.ics.textdb.api.common.Attribute;
+import edu.uci.ics.textdb.api.common.IDictionary;
+import edu.uci.ics.textdb.api.common.IField;
+import edu.uci.ics.textdb.api.common.ITuple;
+import edu.uci.ics.textdb.api.dataflow.IOperator;
+import edu.uci.ics.textdb.common.constants.SchemaConstants;
+import edu.uci.ics.textdb.common.exception.DataFlowException;
+import edu.uci.ics.textdb.common.field.DataTuple;
+import edu.uci.ics.textdb.common.field.IntegerField;
+import edu.uci.ics.textdb.common.field.StringField;
+
+/**
+ * @author Sudeep [inkudo]
+ * 
+ */
+public class DictionaryMatcher implements IOperator {
+
+    private IOperator operator;
+    private IDictionary dict;
+    private String dictValue;
+    private int positionIndex; // next position in the field to be checked.
+    private int fieldIndex; // Index of the next field to be checked.
+    private int spanIndexVal; // Starting position of the matched dictionary
+                              // string
+    private String fieldName;
+    private ITuple dataTuple;
+    private List<IField> fields;
+    private List<Attribute> schema;
```

Comment:
I mentioned this comment before, and I want to repeat it here.  It may be good to create a class called "Schema" which includes a list of attributes. If you agree, please create an issue for this change and do a separate PR.